# SUMMATIVE ASSESMENT 1

### Rationale for Hyphothesis Test



The researcher can conduct a hypothesis test to compare the mean time spent on cell phones by male and female college students. The null hypothesis (H0) would state that the mean times for males and females are equal, while the alternative hypothesis (H1) would state that the mean times are not equal. The rationale for this test is to determine if there is a statistically significant difference between the two groups (male and female)  phone usage, which could reflect behavioral differences between male and female students. An independent two-sample t-test will be used to compare the means, with a significance level of 0.05, allowing the researcher to conclude whether any observed difference is due to random variation or reflects a real difference in the population.

In [4]:
import numpy as np
import scipy.stats as stats
from scipy.stats import shapiro
from numpy.random import randn

males_data = np.array([12,7,7,10,8,10,11,9,9,13,4,9,12,11,9,9,7,12,10,13,11,10,6,12,11,9,10,12,8,9,13,10,9,7,10,7,10,8,11,10,11,7,15,8,9,9,11,13,10,13])

females_data = np.array([11,10,11,10,11,12,12,10,9,9,9,10,8,7,12,9,7,8,9,8,7,7,9,9,12,10,9,13,9,9,10,9,6,12,8,11,8,8,11,12,9,10,11,14,12,7,11,10,9,11])

# null hypothesis
H0 = "The mean number of hours male students and female students spend talking on their cellphones are equal."

# alternative hypothesis
H1 = "The mean number of hours male students and female students spend talking on their cellphones are not equal."

t_stat, p_value = stats.ttest_ind(males_data, females_data)

print("Test statistic (t-value):", t_stat)
print("p-value:", p_value)

alpha = 0.05  # Significance level
if p_value < alpha:
    print("Reject the null hypothesis.")
    print(H1)
else:
    print("Fail to reject the null hypothesis.")
    print(H0)
    



Test statistic (t-value): 0.30394907866566184
p-value: 0.7618111039906375
Fail to reject the null hypothesis.
The mean number of hours male students and female students spend talking on their cellphones are equal.


### Hyphothesis Conclusion and P-Value, What can we conclude?

A p-value greater than 0.05 means that deviation from the null hypothesis is not statistically significant, and the null hypothesis is not rejected.

P-Value provides a measure of how much evidence there is to reject the null hypothesis. The smaller the p-value, the greater the evidence against the null hypothesis.

In practice, the significance level is stated in advance to determine how small the p-value must be to reject the null hypothesis. Because different researchers use different levels of significance when examining a question.

In this Hyphothesis Testing, we choose 95% as our level of confidence). It gives us a conclusion that our data is not statistically significant, it means that the results in the data are explainable by chance alone.

There is not enough evidence to conclude that the mean number of hours male and female students spend talking on their cellphones are different. Based on this analysis, the researcher should maintain the assumption that the means are equal. 


In [19]:
import pandas as pd

males_series = pd.Series(males_data)
females_series = pd.Series(females_data)

print("Descriptive statistics for males:")
print(males_series.describe().round(2))

print("\nDescriptive statistics for females:")
print(females_series.describe().round(2))


Descriptive statistics for males:
count    50.00
mean      9.82
std       2.15
min       4.00
25%       9.00
50%      10.00
75%      11.00
max      15.00
dtype: float64

Descriptive statistics for females:
count    50.00
mean      9.70
std       1.78
min       6.00
25%       9.00
50%       9.50
75%      11.00
max      14.00
dtype: float64


###  95% confidence interval for the population mean of each gender category and for the difference between the means of the two populations

In [14]:
def confidence_interval(data, confidence=0.95):
    n = len(data)
    mean = np.mean(data)
    sem = stats.sem(data) 
    critical_value = stats.t.ppf((1 + confidence) / 2, n - 1)  
    margin_of_error = critical_value * sem
    return mean - margin_of_error, mean + margin_of_error

# Calculate confidence intervals for both groups
males_ci = confidence_interval(males_data)
females_ci = confidence_interval(females_data)

# Calculate confidence interval for the difference between means
mean_difference = np.mean(males_data) - np.mean(females_data)
pooled_std_dev = np.sqrt((np.std(males_data, ddof=1)**2 / len(males_data)) + (np.std(females_data, ddof=1)**2 / len(females_data)))
critical_value_diff = stats.t.ppf((1 + 0.95) / 2, len(males_data) + len(females_data) - 2)
margin_of_error_diff = critical_value_diff * pooled_std_dev
diff_ci = (mean_difference - margin_of_error_diff, mean_difference + margin_of_error_diff)

print(f"95% CI for Males: {males_ci}")
print(f"95% CI for Females: {females_ci}")
print(f"95% CI for the difference in means: {diff_ci}")



95% CI for Males: (9.207794314064703, 10.432205685935298)
95% CI for Females: (9.19535558679254, 10.20464441320746)
95% CI for the difference in means: (-0.6634736514965714, 0.9034736514965734)


### Key Findings:


#### Test Results:

The t-value of 0.304 and a p-value of 0.762 indicate that there is no statistically significant difference between the means of cellphone usage for male and female students. We fail to reject the null hypothesis, suggesting that both groups spend similar amounts of time on their cellphones.

#### Descriptive Statistics:

The means for males (9.82 hours) and females (9.70 hours) are quite close, with the standard deviations also showing a relatively small variation in each group (2.15 for males and 1.78 for females).

#### Confidence Intervals (CIs):

The 95% CIs for both males (9.21, 10.43) and females (9.20, 10.20) overlap, reinforcing the conclusion that there is no significant difference in means. The CI for the difference in means (-0.66, 0.90) also includes zero, which indicates no significant effect.

### Do you see a need for larger sample sizes and more testing with the time spent on cell phones? Discuss.

As discussed, the results indicate that there is no statistically significant difference in cellphone usage between male and female students,  by the p-value of 0.761. This suggests that a larger sample size is necessary to detect any potential differences between groups accurately.

The current sample size of 50 for each gender may not capture the variability in cellphone usage adequately, the standard deviations: 2.15 for males and 1.78 for females. These values indicate some variability in how much time students spend on their phones, which larger samples could help clarify.

A larger sample size would not only enhance the precision of the estimates but also reduce the likelihood of sampling errors, providing a more accurate representation of the population. However, we must also consider practical constraints, such as time and resources, when aiming for a larger sample.

Additionally, factors like age, study load, and access to technology could significantly impact cellphone usage patterns. By including these variables in future studies, we could gain deeper insights into the behaviors affecting cellphone usage among students. Overall, the combination of a larger sample and consideration of these factors could lead to a more comprehensive understanding of cellphone usage differences between genders.

### Report including the testing of the assumptions for two independent samples t-test.


#### Assumptions:

- Random samples
- Independent observations
- The population of each group is normally distributed.
- The population variances are equal.

In [20]:

#  Test for normality
shapiro_males = stats.shapiro(males_data)
shapiro_females = stats.shapiro(females_data)

# Test for homogeneity of variances 
levene_test = stats.levene(males_data, females_data)

# 3. Perform the t-test
t_test_result = stats.ttest_ind(males_data, females_data)


print("Shapiro-Wilk Test for Normality:")
print(f"  Males: Statistic = {shapiro_males.statistic:.4f}, p-value = {shapiro_males.pvalue:.4f}")
print(f"  Females: Statistic = {shapiro_females.statistic:.4f}, p-value = {shapiro_females.pvalue:.4f}")
print("\nLevene's Test for Homogeneity of Variances:")
print(f"  Statistic = {levene_test.statistic:.4f}, p-value = {levene_test.pvalue:.4f}")
print("\nIndependent Samples T-Test:")
print(f"  Statistic = {t_test_result.statistic:.4f}, p-value = {t_test_result.pvalue:.4f}")


Shapiro-Wilk Test for Normality:
  Males: Statistic = 0.9747, p-value = 0.3540
  Females: Statistic = 0.9639, p-value = 0.1292

Levene's Test for Homogeneity of Variances:
  Statistic = 0.6943, p-value = 0.4067

Independent Samples T-Test:
  Statistic = 0.3039, p-value = 0.7618


- Independence:
     Since these data points represent different individuals in separate groups, this assumption is satisfied.
- Normality:
    The p-values for both the male and female groups are greater than 0.05. This indicates that we fail to reject the null hypothesis of normality for both groups. Thus, we can assume that both samples are normally distributed.
    
    
- Variance:
    The p-value is greater than 0.05, suggesting that we fail to reject the null hypothesis that the variances are equal. Therefore, we can assume that the variances of the two groups (males and females) are homogeneous.
    
    
- T-test:
    The p-value is significantly greater than 0.05, indicating that we fail to reject the null hypothesis of equal means. This suggests that there is no statistically significant difference in the mean number of hours spent talking on cellphones between male and female students.